internal/schema/0.13/data_block.go

```diff
@@ -0,0 +1,48 @@
+package schema
+
+import (
+	"github.com/hashicorp/hcl-lang/lang"
+	"github.com/hashicorp/hcl-lang/schema"
+	"github.com/zclconf/go-cty/cty"
+)
+
+var datasourceBlockSchema = &schema.BlockSchema{
+	Labels: []*schema.LabelSchema{
+		{
+			Name:        "type",
+			Description: lang.PlainText("Data Source Type"),
+			IsDepKey:    true,
+		},
+		{
+			Name:        "name",
+			Description: lang.PlainText("Reference Name"),
+		},
+	},
+	Description: lang.PlainText("A data block requests that Terraform read from a given data source and export the result " +
+		"under the given local name. The name is used to refer to this resource from elsewhere in the same " +
+		"Terraform module, but has no significance outside of the scope of a module."),
+	Body: &schema.BodySchema{
+		Attributes: map[string]*schema.AttributeSchema{
```

Comment:
May not be worth the trouble, but in general the Terraform schema is additive and not breaking. With that being the main case, would be be better to instead mutate/add onto the existing schema definitions you defined in 0.12, instead of copypasting the same schema in multiple places, should there need to be a bugfix to the "base" schema, we will need to make sure every versioned definition get's updated.

Not likely a huge problem now, but could become prone to problems/annoyance later.

Reply:
I was kinda torn between the two approaches and I'm still unsure which one is better.

Having entirely "stacked up" schemas, where the latest schema is basically a combination of all stacks below and modelling it that way up to 0.14 would _mostly_ work.

However there are also arguments for some separation:
 - deprecations will happen, no matter how well you plan - so from that perspective the schema is _not_ strictly additive. e.g. `provider.version` attribute will be removed some future versions
 - readability of the code - the more conditionals are involved, the harder it is to understand what the real latest schema actually is and the more likely it is that someone will misread it

It's fair to say that we _will_ need to change the old schemas soon as we add support for expressions, so to make these changes easier I will try to reduce the repetition. I would really prefer to have a clean `1.0` schema (as we get closer to Terraform v1.0) in its own package though for the reasons mentioned above.